packages/core/admin/admin/src/content-manager/pages/EditView/components/DocumentActions.tsx

```diff
@@ -80,57 +104,324 @@ const DocumentActions = ({ actions }: DocumentActionsProps) => {
   }
 
-  const isMenuDisabled = restActions.every((action) => action.disabled);
-
   return (
     <Flex direction="column" gap={2} alignItems="stretch" width="100%">
       <Flex gap={2}>
-        <Button
-          flex={1}
-          startIcon={primaryAction.icon}
-          disabled={primaryAction.disabled}
-          onClick={primaryAction.onClick}
-          justifyContent="center"
-        >
-          {primaryAction.label}
-        </Button>
-        {restActions.length > 0 ? (
-          <Menu.Root>
-            <Menu.Trigger
-              disabled={isMenuDisabled}
-              size="S"
-              endIcon={null}
-              paddingTop="7px"
-              paddingLeft="9px"
-              paddingRight="9px"
-              variant="secondary"
-            >
-              <More aria-hidden focusable={false} />
-              <VisuallyHidden as="span">
-                {formatMessage({
-                  id: 'content-manager.containers.edit.panels.default.more-actions',
-                  defaultMessage: 'More actions',
-                })}
-              </VisuallyHidden>
-            </Menu.Trigger>
-          </Menu.Root>
-        ) : null}
+        <DocumentActionButton {...primaryAction} variant={primaryAction.variant || 'default'} />
+        {restActions.length > 0 ? <DocumentActionsMenu actions={restActions} /> : null}
       </Flex>
       {secondaryAction ? (
-        <Button
-          startIcon={secondaryAction.icon}
-          disabled={secondaryAction.disabled}
-          onClick={secondaryAction.onClick}
-          justifyContent="center"
-          variant="secondary"
-        >
-          {secondaryAction.label}
-        </Button>
+        <DocumentActionButton
+          {...secondaryAction}
+          variant={secondaryAction.variant || 'secondary'}
+        />
       ) : null}
     </Flex>
   );
 };
 
+/* -------------------------------------------------------------------------------------------------
+ * DocumentActionButton
+ * -----------------------------------------------------------------------------------------------*/
+
+interface DocumentActionButtonProps extends Action {}
+
+const DocumentActionButton = (action: DocumentActionButtonProps) => {
+  const [dialogId, setDialogId] = React.useState<string | null>(null);
+  const toggleNotification = useNotification();
+
+  const handleClick = (action: Action) => (e: React.MouseEvent) => {
+    if (action.onClick) {
+      action.onClick(e);
+    }
+
+    if (action.dialog) {
+      switch (action.dialog.type) {
+        case 'notifcation':
+          toggleNotification({
+            title: action.dialog.title,
+            message: action.dialog.content,
+            type: action.dialog.status,
+            timeout: action.dialog.timeout,
+            onClose: action.dialog.onClose,
+          });
+          break;
+        case 'dialog':
+        case 'modal':
+          e.preventDefault();
+          setDialogId(action.id);
+      }
+    }
+  };
```

Comment:
Wouldn't we want a dialog or modal being present to always stop the onClick
(e.g. don't actually save the doc if theres a prompt first)

There's the preventDefault here but wouldn't the onClick still fire from L135?